app/src/main/java/edu/gatech/spacebarz/buzzshelter/model/Shelter.java

```diff
@@ -1,115 +1,116 @@
 package edu.gatech.spacebarz.buzzshelter.model;
 
-import android.location.Location;
-
-/**
- * Created by mayaholikatti on 2/19/18.
- */
-
 public class Shelter {
+    public Shelter() {};
 
-    public enum Age {
-        FamiliesWithNewborns, Children, YoungAdult, Anyone;
-    }
-
-    public enum Gender {
-        MALE, FEMALE;
+    public Shelter(String id, String n, String capS, String res, String addr, String note, String ph, int capN, double lat, double lon, Gender gen, boolean vet) {
+        uid = id;
+        name = n;
+        capacityStr = capS;
+        restrictions = res;
+        address = addr;
+        notes = note;
+        phone = ph;
+        capacityNum = capN;
+        this.lat = lat;
+        this.lon = lon;
+        gender = gen;
+        veteran = vet;
     }
 
-    private int capacity;
-    private String shelterName;
-    private boolean hasFamily;
-    private boolean isVet;
-    private Gender gender;
-    private Age age;
-    private String key;
-    private String phoneNum;
-    private String specialNotes;
-    private Location address;
+    public boolean equals(Object o) {
+        if (o.getClass() != Shelter.class)
+            return false;
 
-    /** Required for use with Firebase, should not be used by the actual app */
-    public Shelter() {
+        Shelter temp = (Shelter) o;
 
+        return temp.getUID().equals(uid) && temp.getName().equals(name) && temp.getCapacityStr().equals(capacityStr) &&
```

Comment:
This can be as simple as a comparison of the UIDs since no two shelters can have the same UID

Reply:
This is to account for changes to individual fields in the CSV in the uploader, I just copied the file. We can change the comparison method for the app, but we may want to keep this if we do offline caching with selective updates.

Reply:
Maybe add a compareUID method if that functionality is going to be used a lot.

Reply:
My worry is that we will get two shelters with the same UID but different attributes in the system, which is not ideal. I would be more for having isEqual compare UIDs but adding an isIdentical or doFieldsMatch method

Reply:
We should have upload safeguards that prevent ID duplication. Not sure how to do this or if it is possible with Firebase.

I'm fine as long as we have both functionalities.

Reply:
Agreed, my thought was to use just the UID for `equals()` so that the `contains()` method and similar will work properly, but agree both are nice. I can add that